src/Components/AdvancedSearch/Internal/AdvancedSearchResultDetailsList/AdvancedSearchResultDetailsList.tsx

```diff
@@ -71,39 +77,46 @@ const AdvancedSearchResultDetailsList: React.FC<IAdvancedSearchResultDetailsList
         switch (column.key) {
             case 'properties':
                 return (
-                    <IconButton
-                        iconProps={{ iconName: 'EntryView' }}
-                        title={t('advancedSearch.inspectProperty')}
-                        ariaLabel={t('advancedSearch.inspectProperty')}
-                        onClick={(event) => {
-                            event.stopPropagation();
-                        }}
-                        className={'cb-scenes-action-button'}
+                    <PropertyInspectorCalloutButton
+                        twinId={`${currentTwin?.$dtId}`}
+                        adapter={adapter}
                     />
                 );
             default:
                 return <span>{fieldContent}</span>;
         }
     };
+    const selection = new Selection({
+        getKey(item, index?) {
+            return item + index;
+        },
+        onSelectionChanged: () => {
+            setTwin(selection.getSelection());
```

Comment:
What would you think of isntead of storing it to state, just passing the value of `selection.getSelection` directly to the `onTwinSelection` callback. Maybe I'm missing something, but I am not seeing that that state is used for anything else and you can't rely on this "setState" happening before the next line executes (in fact I don't think it ever will since set state triggers a render and it will finish the current render loop first) This is likely the cause of the other bug I mentioned where the inspector does not have the right value.

Reply:
yeah you're right, the currentTwin state was not needed so yeah selection.getSelection() can be passed directly to onTwinSelection